### DISCUSSION TITLE
What Player Event returns the notes being played.

### DISCUSSION THREAD

**Discussion by eyeke04** [cifkao/html-midi-player#44] [General]
Hi Cifkao, thank you so much for this amazing gift. I want to be able to display the notes played on a custom HTML and CSS model of the instrument (much like the midi-visualizer). Please which of the player event returns the current notes being played? I've tried ``` player.addEventListener('note', listener)``` and ``` player.addEventListener('playing', listener)```but none of them works.

How do I achieve that?

![Uploading Screenshot 2022-04-27 at 01.06.33.png…]()

**Comment by cifkao** [comment]
Hi, it is the `note` event as documented. It's a [`CustomEvent`](https://developer.mozilla.org/en-US/docs/Web/API/CustomEvent), so you have to access its `detail` property to get the actual note, e.g.:
```javascript
player.addEventListener("note", (event) => console.log(event.detail.note));
```

If it's still not working, please share your code (your screenshot failed to upload).

**Reply by eyeke04** [reply]
Thank you for the response. The `note` event isn't firing at all. The `load` event is firing but when I click the play button, it throws this warning `The AudioContext is "suspended". Invoke Tone.start() from a user action to start the audio.`, continues playing but doesn't fire the `note` event. Actually no other event fires after that.

```
window.addEventListener('DOMContentLoaded', function() {
            var player = document.querySelector('midi-player');
            player.addEventListener('load', (e)=>{
                console.log(e.detail);
           });
            player.addEventListener('note', (e)=>{
                console.log(e.detail);
            });
           
  });
  
  ```

**Reply by cifkao** [reply]
Interesting, for me it does work. Would you mind sharing a complete example e.g. on [codepen](https://codepen.io/), including the HTML? (Your code works when I paste it into [my example](https://codepen.io/cifkao/pen/WNwpLzL).) Also, what is your browser (version) and OS? Have you tried it in other browsers?

**Reply by eyeke04** [reply]
Oh wow, I'm using Mac Monterey 12.2.1. I tried it on Safari and it works perfectly but it's still failing on Chrome v 101.0.4951.41. And it throws all these warning on chrome that aren't on Safari.

<img width="1495" alt="Screenshot 2022-04-27 at 12 37 45" src="https://user-images.githubusercontent.com/10612514/165510545-2ad72c40-eec2-413a-8abd-8270c9cd4730.png">

**Reply by cifkao** [reply]
Hmmm... working for me in Chrome 101.0.4951.41 on macOS Monterey 12.3.1... (But I also get these warnings, they seem to be normal.)

**Reply by eyeke04** [reply]
wow, ok. Let me figure out what is happening. Thank you. Also thank you for being so responsive. Do you have a community for this that I can join or it's just here?

**Reply by eyeke04** [reply]
@cifkao I've found the problem. It's this webmidi script `<script src="https://cdn.jsdelivr.net/npm/webmidi@next/dist/iife/webmidi.iife.js"></script>`. I use it to read input from Midi devices. When it's present, the eventListener doesn't bind. But this is happening only on chrome
